api/v1alpha1/extension_types.go

```diff
@@ -20,7 +20,17 @@ import (
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 )
 
-type ExtensionPackage struct {
+// Although this looks like a bool, there might
+type ExtensionPaused string
+
+const (
+	// Pause resolution of this Extension
+	ExtensionPausedTrue ExtensionPaused = "True"
+	// Peform resolution of this Extension
+	ExtensionPausedFalse ExtensionPaused = "False"
```

Comment:
Sorry, should have been more specific earlier. What about something with:
- `Paused` and `Active` enum values
- `Active` would be default.

In the future, this would give us space to easily introduce another meaningful enum value.

Reply:
The spec named the field `paused`. Having the field and a value be the same makes little sense. Do you have a suggestion for new name?

Reply:
I really like paused true/false and I'm not sure I'm convinced that in the future we'd want to expand paused with more values... but some brainstorming if we want to:

managed: active/paused
management.state: active/paused
reconcile: active/paused
reconciliation: active/paused

Reply:
I wrote this as `resolution` rather than reconciliation, so thinking `managed` might be more appropriate.

Reply:
I don't have a really strong opinion here. What I want to avoid is a situation where there is some other knob that we decide we need to add later that isn't completely orthogonal to `spec.paused`. I can't think of anything off hand, though. This does seem fairly cut and dry.

Not exactly the same, but the OLMv0 CatalogSource API has some fields that become inert based on the value of some other fields. Perhaps the main issue there is that we didn't add validation at the time that would prevent setting a field if something else made it inert.